c/planarityTestGraphFunctionality.c

```diff
@@ -111,3 +212,130 @@ int transformString(graphP theGraph, char *inputStr)
 	
 	return gp_ReadFromString(theGraph, inputStr);
 }
+
+/*
+	TEST ALL GRAPHS IN .G6
+*/
+
+int testAllGraphs(graphP theGraph, char command, char *inputStr, strOrFileP testOutput)
+{
+	int exitCode = OK;
+
+	char *errorStr = NULL;
+
+	graphP copyOfOrigGraph = NULL;
+	int embedFlags = GetEmbedFlags(command);
+	int numGraphsRead = 0, numOK = 0, numNONEMBEDDABLE = 0;
+
+	G6ReadIterator *pG6ReadIterator = NULL;
+	exitCode = allocateG6ReadIterator(&pG6ReadIterator, theGraph);
+
+	if (exitCode != OK)
+	{
+		ErrorMessage("Unable to allocate G6ReadIterator.\n");
+		return exitCode;
+	}
+
+	exitCode = beginG6ReadIterationFromG6String(pG6ReadIterator, inputStr);
+
+	if (exitCode != OK)
+	{
+		ErrorMessage("Unable to begin .g6 read iteration.\n");
+
+		exitCode = freeG6ReadIterator(&pG6ReadIterator);
+
+		if (exitCode != OK)
```

Comment:
If the begin failed but then the free succeeds, this will return an OK exitCode when it shouldn't. It'd be better to just say `if (freeG6ReadIterator(...) != OK) then error message` because the exitCode is not reassigned from the original failure code produced by the begin iteration call. Or, just freeG6ReadIterator() but don't bother to check if it fails and print an error because the begin already failed.